### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "undefined"

### DISCUSSION THREAD

**Discussion by alexfdiaz** [community/community#9284] [Copilot Conversations]
Does anyone have this problem? How do I fix it?
I just installed the Github Copilot extension in VSCode, signed in with my account but I can't use it. Any ideas?

**Comment by DaniGalicia** [comment]
Just close VSCode and try to activation again. It works.

**Comment by fklassen** [comment]
I am having the same issue since I upgraded my Linux VS Code to 1.63.2. My macOS installation is at the same version.

I also have issues with Live Share authentication via GitHub. My GitHub extension works (can pull/push).

**Comment by EDBZ** [comment]
Hello, same problem here, version 1.63.2 on Windows_NT x64 10 operating system.
Here is the console feedback : 

```
[INFO] [default] [2022-01-12T15:41:52.475Z] Experimentation service initialized  
[INFO] [default] [2022-01-12T15:41:52.853Z] Stringified features: {}  
[INFO] [default] [2022-01-12T15:41:52.853Z] Feature for copilotaa: false  
[INFO] [default] [2022-01-12T15:41:52.853Z] Feature for copilotincludeprompt: none  
[INFO] [default] [2022-01-12T15:41:52.853Z] Feature for copilotset1: undefined  
[INFO] [default] [2022-01-12T15:41:52.853Z] Feature for copilotghgh: false  
[INFO] [default] [2022-01-12T15:41:52.853Z] Feature for copilotfim: false  
[ERROR] [default] [2022-01-12T15:42:10.506Z] GitHub Copilot could not connect to server. Extension activation failed: "undefined"  
```

**Comment by davecheney** [comment]
Are you using code or vscodium?

**Comment by davecheney** [comment]
Is it possible you have installed code, or vscodium rather than vscode? Copilot only supports vscode, not their unbranded variants often packaged by linux distributions.

**Reply by EDBZ** [reply]
I have a VSCode installation from [https://code.visualstudio.com/Download](https://code.visualstudio.com/Download).
I just reinstalled for security, the problem persists

**Reply by davecheney** [reply]
Thanks for confirming.

**Comment by davecheney** [comment]
All: would you please try the latest build of copilot-nightly https://marketplace.visualstudio.com/items?itemName=GitHub.copilot-nightly and if the problem continues, please provide the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Comment by EDBZ** [comment]
Hello, by following the process indicated, here is the output obtained:
```
[ERROR] [default] [2022-01-17T08:16:16.820Z] GitHub Copilot could not connect to server. Extension activation failed: "authenticationrequired"
```

**Reply by davecheney** [reply]
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Comment by guilhem-martin** [comment]
Hello,
you may need to enable/disable the proxy since Copilot needs to discuss with the server.
In the *settings.json* of vs code, set the following variables depending on your configuration or remove them if you don't use any proxy: 
```
"http.proxy": "your proxy url"
"https.proxy": "your proxy url"
```

**Comment by yezert** [comment]
[ERROR] [default] [2022-04-08T04:02:05.257Z] GitHub Copilot could not connect to server. Extension activation failed: "Timed out waiting for authentication provider to register"

**Comment by maxisam** [comment]
GitHub Copilot could not connect to server. Extension activation failed: "Unexpected end of JSON input"
 
 Now I got this

**Reply by AlePro12** [reply]
Same 
![Captura de Pantalla 2022-06-21 a la(s) 1 06 14 p m](https://user-images.githubusercontent.com/46333161/174857697-aa1597ec-8070-4be4-b49a-82e523d13bce.png)

**Reply by supermavster** [reply]
Hi @maxisam and @AlePro12 I make a suggest solution here: [https://github.com/orgs/github-community/discussions/19061#discussioncomment-2995893](https://github.com/orgs/github-community/discussions/19061#discussioncomment-2995893)

**Comment by GoulartNogueira** [comment]
For me, the root cause was:

- I had not approved the current workspace as a secure environment.
- Somehow the github cannot login in that scenario
- I recieved the same error message

I tried everything (including resetting all the config)

**How I fixed:**

- Simple change to another workspace.
- Confirm it's a secure workspace
- Automatic login to github
- Copilot working perfectly

**Comment by ItayShallev** [comment]
You probably have **GitHub Authentication disabled**:
https://stackoverflow.com/questions/71007828/github-copilot-could-not-connect-to-server
